jfr-mappers/src/main/java/com/newrelic/jfr/tosummary/BasicGarbageCollectionSummarizer.java

```diff
@@ -0,0 +1,141 @@
+            {
+              add("DefNew");
+              add("G1New");
+              add("ParallelScavenge");
+              add("ParNew");
+              add("PSMarkSweep");
+            }
+          });
+
+  private static final Set<String> MAJOR_GC_NAMES =
+      Collections.unmodifiableSet(
+          new HashSet<String>() {
+            {
+              add("ConcurrentMarkSweep");
+              add("G1Full");
+              add("G1Old");
+              add("ParallelOld");
+              add("SerialOld");
+            }
+          });
+
+  public BasicGarbageCollectionSummarizer() {
+    this(Instant.now().toEpochMilli());
+  }
+
+  public BasicGarbageCollectionSummarizer(long startTimeMs) {
+    this(
+        startTimeMs,
+        new SimpleDurationSummarizer(startTimeMs, DEFAULT_CLOCK, "duration"),
+        new SimpleDurationSummarizer(startTimeMs, DEFAULT_CLOCK, "duration"));
+  }
+
+  public BasicGarbageCollectionSummarizer(
+      long startTimeMs,
+      SimpleDurationSummarizer minorGcDurationSummarizer,
+      SimpleDurationSummarizer majorGcDurationSummarizer) {
+    this.startTimeMs = startTimeMs;
+    this.minorGcDurationSummarizer = minorGcDurationSummarizer;
+    this.majorGcDurationSummarizer = majorGcDurationSummarizer;
+  }
+
+  @Override
+  public String getEventName() {
+    return EVENT_NAME;
+  }
+
+  @Override
+  public void accept(RecordedEvent ev) {
+    final String name = ev.getValue("name");
+    if (name != null) {
+      if (MINOR_GC_NAMES.contains(name)) {
+        minorGcEndTimeMs = ev.getStartTime().toEpochMilli();
+        minorGcDurationSummarizer.accept(ev);
+        minorGcCount++;
+      } else if (MAJOR_GC_NAMES.contains(name)) {
+        majorGcEndTimeMs = ev.getStartTime().toEpochMilli();
+        majorGcDurationSummarizer.accept(ev);
+        majorGcCount++;
+      } else
+        // Ignore events with GC name: GCNameEndSentinel, N/A, Shenandoah, Z or anything unexpected
+        logger.warn("Ignoring unsupported " + EVENT_NAME + " event: " + name);
+    }
+  }
+
+  @Override
+  public Stream<Summary> summarize() {
+    Attributes attr = new Attributes();
+    Summary minorGcDuration =
+        new Summary(
+            "jfr.GarbageCollection.minorDuration",
+            minorGcCount,
+            minorGcDurationSummarizer.getDurationMillis(),
+            minorGcDurationSummarizer.getMinDurationMillis(),
+            minorGcDurationSummarizer.getMaxDurationMillis(),
+            startTimeMs,
+            minorGcEndTimeMs,
+            attr);
+
+    Summary majorGcDuration =
+        new Summary(
+            "jfr.GarbageCollection.majorDuration",
+            majorGcCount,
+            majorGcDurationSummarizer.getDurationMillis(),
+            majorGcDurationSummarizer.getMinDurationMillis(),
+            majorGcDurationSummarizer.getMaxDurationMillis(),
+            startTimeMs,
+            majorGcEndTimeMs,
+            attr);
+    return Stream.of(minorGcDuration, majorGcDuration);
```

Comment:
@kittylyst Is this the correct approach, always creating both metrics? Or should there be logic that only returns a stream of summarizers that have a count > 0? 

On the one hand, this approach means that both metrics will always have _some_ value and nice continuous lines in the charts. On the other, we would potentially be sending up metrics with the default value of 0 for count/sum/min/max.

I'd love to get some feedback on which approach is more desirable.

Reply:
@kittylyst  IMO, from a useability perspective, `0` and a flat line is information that we are monitoring major GC events and that event hasn't happened yet.  That is reassuring and removes doubt "Is jfr recording major gc events??"    2️⃣ 💴

Reply:
👍 to @XiXiaPdx comments